homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -0,0 +1,47 @@
+"""Code to handle a Xiaomi Gateway."""
+import logging
+
+from miio import DeviceException, gateway
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConnectXiaomiGateway:
+    """Class to async connect to a Xiaomi Gateway."""
+
+    def __init__(self, hass):
```

Comment:
API wise it would be better to add `cls` parameter here, and pass the `Gateway` as a parameter to the class. This will make it generic for other platforms to use.